### PR TITLE
core: Add RouteConfigurationsBuilder before regular RoutesBuilder whe…

### PR DESCRIPTION
**Pull request by aldettinger**
…n camel main is disabled #2978

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java [hunk 1]
```diff
@@ -90,6 +93,8 @@ public void customize(RuntimeValue<CamelContext> context, RuntimeValue<CamelCont
         return new RuntimeValue((CamelContextCustomizer) context -> context.setShutdownStrategy(new NoShutdownStrategy()));
     }
```
Comment (aldettinger) [inline extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java:95]: I've marked those recorded methods as deprecated. Maybe it's even ok to remove them ? I don't know what we have done in the past for that kind of situations.

Comment (zbendhiba) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java:95]: If it's for internal use only, I'd vote for removing them. 
If the user can use them, I'd say maybe deprecate and remove when introducing 3.14, the next LTS ?

Comment (jamesnetherton) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java:95]: Agreed - I think for internal stuff in the past like recorder methods, we have removed them without deprecation. It's not public API.

Comment (aldettinger) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java:95]: Okay I've removed the useless methods. Thanks all.

File: extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelContextProcessor.java [hunk 2]
```diff
@@ -165,17 +165,12 @@ public CamelRuntimeBuildItem runtime(
             List<CamelRuntimeTaskBuildItem> runtimeTasks,
             CamelConfig config) {
 
-        for (CamelRoutesBuilderClassBuildItem item : routesBuilderClasses) {
-            // don't add routes builders that are known by the container
-            if (containerBeans.getClasses().contains(item.getDotName())) {
-                continue;
-            }
-
-            recorder.addRoutes(context.getCamelContext(), item.getDotName().toString());
```
Comment (aldettinger) [inline extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelContextProcessor.java:174]: I've spread the logic from the loop above in the code block below. So, the logic is now executed only when config.routesDiscovery.enabled = true.

However, to my current understanding, when config.routesDiscovery.enabled = false, then the routesBuilderClasses list is empty. As such, the logic in the loop above was not executed when config.routesDiscovery.enabled = false.

**Review by zhfeng — APPROVED** [review]
LGTM and well done !

**Review by zbendhiba — APPROVED** [review]

**Pull request merged**